### DISCUSSION TITLE
[Feature request] Passing alert label value to rendered dashboard/panel variable

### DISCUSSION THREAD

**Discussion by GedPL** [grafana/grafana#56004] [Alerting]
Hello,
I’m facing a problem with rendering images for multi dimensional alert rules.
Let my elaborate on below example.

In my case data source is Google Cloud Monitoring.
I have dashboard containing graphs showing some VM Instances metrics (CPU, memory, etc.) Each graph is grouped by instance name. There are two variables for this dashboard:
project - as it’s used for a few GCP projects
host - as I want to filter graphs for specific instance from time to time.

![image](https://user-images.githubusercontent.com/31762538/193016411-88638c7d-7667-4b4f-b0a9-469cbd59757c.png)

I also created multi dimensional single alert rule for VM instances CPU usage.
This one is associated to “CPU usage of each VM Instance” graph shown above, using DashoardID and PanelID.
Now, assuming I have 10 VM instances I may receive one email with 1-10 notifications when alert is triggered.

![image](https://user-images.githubusercontent.com/31762538/193016745-04eb7956-2a64-4362-9765-a22943549343.png)

And each of this notification will have the same rendered picture attached as it is not possible to pass VMs instance name to rendered graph as a variable.

Another thing. As I’ve mentioned above I’ve “project” variable in my dashboard for viewing VMs for a few projects. Having no option for passing project name to graph variable “project” when rendering I would need to create separate dashboards/graphs for each project. 

So how it should work for multi dimensional alert rules in my opinion ?
If images rendering is enabled and alert rule is correctly associated to dashboard/panel, we should have additional configuration option determining values pairs saying which label in alert rule should be passed to which dashboard variable, for example in my case:
resource.label.project_id → $project
metadata.system_labels.name → $instance

What we’ll achieve this way ?

Using the same dashboard/panel for many projects avoiding multiplying them
Each notification in single alert email would have image attached showing CPU usage only for related VM instance.
Additionally I would consider adding one more option for multi dimensional alert, let’s call it “Single image” to avoid rendering and attaching many pictures to one email.
How would it work in my use case ?
Regardless of VM instances number alerting high CPu usage I can show it on one picture (based on panel showing CPU usage for each VM instance) as in fact I don’t need to have separate pictures for each alerted VM instance.
In such case I would only to pass only one label to dashboard variable:
resource.label.project_id → $project
for selecting GCP project alert rule is related to and enable “Single image” option.

Summarizing: My understanding is currently image rendering functionality is useful only for single alerts (not multi dimensional) not grouped, but definitely is great one.
Just should be improved a little bit :)
Hoping someone will review this as Feature Request.

Have a nice day !

Ged

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/alerting-squad
